ddtrace/appsec/_trace_utils.py

```diff
@@ -37,7 +36,7 @@ def _asm_manual_keep(span: Span) -> None:
 
 
 def _track_user_login_common(
-    tracer: Tracer,
+    tracer,
```

Comment:
We need to remove this typing to resolve circular imports between the `Tracer` and `AppSecProccessor`.

Reply:
I will add an abstract tracer type to `ddtrace/_trace/types.py` (BaseTracer). This will allow us to use the BaseTracer type here and in the Tracer class. This will break the circular loop.
cc: @avara1986